src/installer/utils.py

```diff
@@ -103,3 +107,49 @@ def copyfileobj_with_hashing(
         size += len(buf)
 
     return hasher.hexdigest(), size
+
+
+def get_launcher_kind():  # pragma: no cover
+    # type: () -> LauncherKind
+    if os.name != "nt":
+        return "posix"
+
+    if "amd64" in sys.version.lower():
+        return "win-amd64"
+    if "(arm64)" in sys.version.lower():
+        return "win-arm64"
+    if "(arm)" in sys.version.lower():
+        return "win-arm"
+    if sys.platform == "win32":
+        return "win-ia32"
+
+    raise NotImplementedError("Unknown launcher kind for this machine")
+
+
+def fix_shebang(data, interpreter):
```

Comment:
I'd call it stream and new_stream -- data implies that it's a bytes/string to me.

Reply:
Same note about closing the BytesIO applies.

Reply:
I made `fix_shebang` a context manager because it is in a weird position where it may either return the original stream or a new one. The caller doesn't know if they should explicitly close the stream we return or not.
Using a context manager allows us to keep the original stream open and close ours if needed.